lib/logstash/devutils/rspec/shared_examples.rb

```diff
@@ -0,0 +1,23 @@
+require 'rspec/wait'
+
+RSpec.shared_examples "an interruptible input plugin" do
+  describe "#stop" do
+    let(:queue) { SizedQueue.new(20) }
+    subject do
+      s = described_class.new(config)
+      s.register
+      s
+    end
+    it "returns from run" do
+      consumer_thread = Thread.new(queue) { |queue| loop { queue.pop } }
+      plugin_thread = Thread.new(subject, queue) { |subject, queue| subject.run(queue) }
+      # the run method is a long lived one, so it should still be running after "a bit"
+      sleep 0.5
```

Comment:
My 2 cents.  For code above (rabbitmq input) the call to subscribe will register a callback block, so the input is ready after subscribe returns.  Unless subscribe never returns until the server is stopped (because control is handed to a thread in the server), in this case the input is ready after the call to new/before the call to subscribe and the input is stopped when the call to subscribe returns